lib/utils.js

```diff
@@ -477,4 +477,311 @@
+     *
+     * @function splunkjs.Utils
+     */
+    root.parallelEach =  async function (vals, fn) {  
+        vals = vals || [];
+        let [err,res] = await root.parallelMap(vals, fn);
+        return err || null;
+    };
+
+    /**
+     * Runs multiple functions (tasks) in series. 
+     * Each task takes the function as a parameter. 
+     * When all tasks have been completed or if an error occurs, the 
+     * function returns the combined results of all tasks in the order
+     * they were run.
+     * 
+     * @example
+     *      
+     *      var keeper = 0;
+     *      let [err, one, two] = awiat Utils.series([
+     *          async function() {
+     *              await Utils.sleep(10);
+     *              console.log(keeper++); // == 0
+     *              return [null, 1];
+     *          },
+     *          function() {
+     *              console.log(keeper++); // == 1
+     *              return [null, 2, 3];
+     *          }]
+     *      );
+     *      console.log(err); // == null
+     *      console.log(one); // == 1
+     *      console.log(two); // == [2, 3]
+     *
+     * @param {Function} tasks An array of functions.
+     * @param {Boolean} fromMap set to true when method call is made from seriesMap function. (optional)
+     *
+     * @function splunkjs.Utils
+     */
+    root.series = async function (tasks, fromMap) {
+        let res = [];
+        if(!root.isArray(tasks)&& root.isFunction(fromMap)){
+            let taskList = [];
+            Object.keys(arguments).forEach(key => {
+                taskList.push(arguments[key]);
+            });
+            tasks = taskList;
+            fromMap = false;
+        }
+        for(let task of tasks) {
+            let result = await task();
+            if(result){
+                if(result[0]){
+                    return [result[0], null];
+                }
+                if(result.length > 2){
+                    res.push(result.slice(1));
+                }else{
+                    res.push(result[1]);
+                }
+            }
+        }
+        return fromMap ? [null, res] : [null, ...res];
+    }
+
+    /**
+     * Runs an asynchronous function (mapping it) over each element in an array, in series.
+     * When all tasks have been completed or if an error occurs, function
+     * returns the resulting array.
+     *
+     * @example
+     *      
+     *      var keeper = 1;
+     *      let [err, vals] = await Utils.seriesMap(
+     *          [1, 2, 3],
+     *          function(val, idx) { 
+     *              console.log(keeper++); // == 1, then 2, then 3
+     *              return [null, val + 1];
+     *          }
+     *      );
+     *      console.log(vals); // == [2,3,4];
+     *
+     * @param {Array} vals An array of values.
+     * @param {Function} fn A function (possibly asynchronous) to apply to each element.
+     *
+     * @function splunkjs.Utils
+     */
+    root.seriesMap = async function (vals, fn) {
```

Comment:
I think these utilities from the Async class are only used by our tests does that seem right? If so maybe we can take an action item after this merges to refactor these out of our tests and remove them

Reply:
as per our discussion regarding scenarios where users might need to execute some action multiple times such as create/delete users or roles or any resource, these utilities would be useful, hence keeping them as is